src/forcemove.h

```diff
@@ -0,0 +1,98 @@
+#pragma once
+#include "move.h"
+#include "energy.h"
+#include "units.h"
+
+namespace Faunus {
+
+namespace Move {
+
+/**
+ * @brief Base class for dynamics integrators
+ *
+ */
+class IntegratorBase {
+  protected:
+    Space &spc;
+    Energy::Energybase &energy;
+    IntegratorBase(Space &spc, Energy::Energybase &energy);
+    virtual ~IntegratorBase() = default;
+
+  public:
+    /**
+     * @brief Move particles by one time step and update their positions, velocities and acting forces
+     * @param velocities
+     * @param forces
+     */
+    virtual void step(PointVector &velocities, PointVector &forces) = 0; // todo shall we return real time progress?
+    virtual void from_json(const json &j) = 0;
+    virtual void to_json(json &j) const = 0;
+};
+
+void from_json(const json &j, IntegratorBase &i);
+void to_json(json &j, const IntegratorBase &i);
+
+/**
+ * @brief Symmetric Langevin velocity-Verlet method (BAOAB)
+ *
+ * The integrator can conduct normal velocity-Verlet, when gamma = 0.
+ */
+class LangevinVelocityVerlet : public IntegratorBase {
+    float dt;    //!< integration time step (picoseconds)
+    float gamma; //!< friction coefficient (inverse picoseconds)
+
+    //! increment particle's position in an A semi-step
+    inline Point positionIncrement(const Point& velocity);
+    //! increment particle's velocity in a B semi-step
+    inline Point velocityIncrement(const Point& force, const double mass);
+    //! apply fluctuation-dissipation to particle's velocity by solving of Ornstein-Uhlenbeck process in an O-step
+    inline Point velocityFluctuationDissipation(const Point& velocity, const double mass);
+  public:
+    LangevinVelocityVerlet(Space &spc, Energy::Energybase &energy, float dt = 1.0, float gamma = 0.0);
```

Comment:
dt -> time_step and gamma -> friction_coefficient

Reply:
done